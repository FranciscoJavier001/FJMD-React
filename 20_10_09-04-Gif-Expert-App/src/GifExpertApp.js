//** Cuando hacemos el build debemos de levantar el servidor ahora con http-server -o, recuerda que antes debes instalar un servidor que levante la app eso se hace con el comando sudo npm install --global http-server */

import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid' //** Es para agregar el category, desde el GifGrid */

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Death Note']) //** Se crea una constante donde vamos a agregar cosas, recuerda este es un hook, setCategories es la instruccion para agregar categorias */

    //setCategories(cats => [...cats, 'HunterXHunter']); /** El UseState me dio la habilidad de hacerlo directamente con el setCategories */

    /** Para crear nuevos componentes en React debemos de crear una carpeta en "src" llamada components y ponerle un nombre descriptivo como "AddCategory" y escribimos rafc y ahi escribimos lo que tengamos que escribir y se muestra en el dom, luego agrego el componente  */

    return ( 
        <>
        {/* Esta es la tarea que dejo */}
            <h2>GifExpertApp</h2> 
            {/* Siempre asegurate que se importe, este ya se importo arriba */}
            {/* setCategories es la instruccion para agregar categorias */}
            {/* Podemos pasarle caegorias por aqui, es decir, le pasamos la referencia a setCategories */}
            <AddCategory setCategories={setCategories}/>
            <hr/>

            {/* Una lista ordenada */}
            <ol>
                {
                    categories.map(category => ( /** El Map barre el arreglo que ahora se llama category, porque esto es una funcion de flecha que ya inicializamos y podemos mandar llamar cuando necesitemos con el return */
                        // GifGrid es el componente que hace la peticion "http" al servidor, para que regrese un objeto definido como category en el GifGrid, que esta evaluando desde el GifGrid, y este necesita el key
                        <GifGrid  
                        key={category}
                        category={category}
                        />
                        // <li key={category}>{category}</li> /** Muestra el arreglo con cada cosa que esta ahi dentro */
                    ))
                }
            </ol>

        </>
    )
}
