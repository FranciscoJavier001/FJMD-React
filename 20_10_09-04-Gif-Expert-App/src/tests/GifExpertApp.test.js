import React from 'react' //** Importamos React para las pruebas */
import '@testing-library/jest-dom' //** Para evitar errores de escritura */
import {shallow} from 'enzyme'  //** Importamos Shallow desde enzyme */
import { GifExpertApp } from '../GifExpertApp' //** Importamos el componente al cual le vamos a hacr la prueba */

describe('Pruebas en el <GifExpertApp/>', () => {
    
    test('Debe de mostrarse corectamente', () => {

        const wrapper = shallow(<GifExpertApp/>) //** Es por default pasadole por argumento el el archivo que voy a evaluar */
        
        expect(wrapper).toMatchSnapshot(); //** Esto hace el snapshot */
    });

    test('Debe de mostrar una lista de categorias', () => { //** Le mandamos archivos desde aqui para ver cuantos espacios tiene */
        const categories = ['Death Note', 'Vegeta'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />); //** Le mandamos las categories como argumento */

        expect(wrapper).toMatchSnapshot(); //** Creaos el snapchot */
        expect(wrapper.find('GifGrid').length).toBe(categories.length); //** Aqui vemos cuantos arreglos tiene y lo comparamos para ver si son los mismos de las categories */
    })
    
})