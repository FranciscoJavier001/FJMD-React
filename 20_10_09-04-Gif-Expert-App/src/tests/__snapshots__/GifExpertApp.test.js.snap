// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Death Note"
      key="Death Note"
    />
    <GifGrid
      category="Vegeta"
      key="Vegeta"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el <GifExpertApp/> Debe de mostrarse corectamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
